types/bool.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+type (
+	Bool bool
+)
+
+var True = Bool(true)
+var False = Bool(false)
+
+func NewBool(b bool) Bool    { return Bool(b) }
+func (_ Bool) Kind() Kind    { return KindBool }
+func (b Bool) IsTrue() bool  { return bool(b) }
+func (b Bool) IsFalse() bool { return !bool(b) }
+func (b Bool) ToBool() Bool  { return b }
+func (b Bool) ToNumber() Number {
+	if b {
+		return NewNumber(1)
+	}
+	return NewNumber(+0)
```

Comment:
+0 ? any reason for the "+" ?

Reply:
http://es5.github.io/#x9.3

Just one more floating point IEEE stuff I dont want to read about. ECMAScript really has meanings for -0 e +0 (I think in the same way as IEEE xxx) but I do know Go does not has a meaning for -0. But just implemented *verbatim* from ES5 specification.